src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,636 @@
+        return transactionDTO;
+    }
+    
+    protected KualiDecimal convertCostStringToKualiDecimal(String costString) {
+        KualiDecimal cost = KualiDecimal.ZERO;
+        try {
+            double costDouble = Double.valueOf(costString);
+            final double oneCent = .01;
+            if (costDouble >= oneCent) {
+                cost = new KualiDecimal(costDouble);
+            }
+        } catch (Throwable e) {
+            LOG.error("convertCostStringToKualiDecimal() Unable to convert " + costString + " to a double.", e); 
+        }
+        return cost;
+    }
+    
+    protected AmazonBillingCostCenterDTO convertCostCenterToAmazonBillingCostCenterDTO(String costCenter) throws Exception {
+        AmazonBillingCostCenterDTO costCenterDTO = new AmazonBillingCostCenterDTO();
+        
+        if (StringUtils.contains(costCenter, KFSConstants.DASH)) {
+            String[] costCenterArray = StringUtils.splitPreserveAllTokens(costCenter, KFSConstants.DASH);
+            if (costCenterArray.length == 2) {
+                costCenterDTO.setAccountNumber(costCenterArray[0]);
+                costCenterDTO.setSubAccountNumber(costCenterArray[1]);
+            } else if (costCenterArray.length == 7) {
+                costCenterDTO.setChartCode(costCenterArray[0]);
+                costCenterDTO.setAccountNumber(costCenterArray[1]);
+                costCenterDTO.setSubAccountNumber(costCenterArray[2]);
+                costCenterDTO.setObjectCode(costCenterArray[3]);
+                costCenterDTO.setSubObjectCode(costCenterArray[4]);
+                costCenterDTO.setProjectCode(costCenterArray[5]);
+                costCenterDTO.setOrgReferenceId(costCenterArray[6]);
+            } else {
+                throw new Exception("There was an incorrect number of elements in the Cost Center");
+            }
+        } else {
+            costCenterDTO.setAccountNumber(costCenter);
+        }
+        
+        return costCenterDTO;
+    }
+    
+    private void updateTransactionDTOChartAccountSubAccount(AmazonBillingDistributionOfIncomeTransactionDTO transactionDTO, AmazonBillingCostCenterDTO costCenterDTO,
+            String amazonDetailKFSAccount) {
+        Account account;
+        if (StringUtils.isNotBlank(costCenterDTO.getAccountNumber())) {
+            account = buildAccount(costCenterDTO.getAccountNumber());
+        } else {
+            account = buildAccount(amazonDetailKFSAccount);
+        }
+        if (validAccount(account)) {
+            transactionDTO.setAccountNumber(account.getAccountNumber());
+            updateTansactionDTOChart(transactionDTO, costCenterDTO, account);
+            updateTransactionDTOSubAccount(transactionDTO, costCenterDTO, account);
+        } else {
+            LOG.info("updateTransactionDTOChartAccountSubAccount() Invalid account.");
+            transactionDTO.setTransactionInputError(true);
+        }
+    }
+    
+    private Account buildAccount(String accountNumber) {
```

Comment:
This method name is rather misleading since it is only retrieving an account rather than building one based on the input data